ansible_runner/config/_base.py

```diff
@@ -153,13 +155,17 @@ def _prepare_env(self, runner_mode='pexpect'):
                     self.passwords = self.passwords.update(self.loader.load_file('env/passwords', Mapping))
                 else:
                     self.passwords = self.passwords or self.loader.load_file('env/passwords', Mapping)
-                self.expect_passwords = {
-                    re.compile(pattern, re.M): password
-                    for pattern, password in iteritems(self.passwords)
-                }
             except ConfigurationError:
                 debug('Not loading passwords')
                 self.expect_passwords = dict()
+            finally:
+                if self.passwords:
+                    self.expect_passwords = {
+                        re.compile(pattern, re.M): password
+                        for pattern, password in iteritems(self.passwords)
+                    }
+                else:
+                    self.expect_passwords = dict()
```

Comment:
this looks like a real bug fix :+1:

Reply:
Wouldn't this actually now cause an issue if the re.compile fails and raises an exception? Before you would fall into the except but now you will get a stack.

Reply:
Problems there wouldn't raise a `ConfigurationError` so it wouldn't be caught anyway.